pickLayer/core/set_active_layer_tool.py

```diff
@@ -99,15 +103,14 @@ def set_active_layer_using_closest_feature(
     def _get_identify_results(
         self, location: QgsPointXY, search_layer_ids: Optional[List[str]] = None
     ) -> List[QgsMapToolIdentify.IdentifyResult]:
-        layer_ids = self.search_layer_ids or []
-        if search_layer_ids is not None:
-            layer_ids = search_layer_ids
-
-        layers = [
-            layer
-            for layer in QgsProject.instance().mapLayers().values()
-            if isinstance(layer, QgsVectorLayer) and layer.id() in layer_ids
-        ]
+        if search_layer_ids:
+            layers = []
+            for layer_id in search_layer_ids:
+                layer = QgsProject.instance().mapLayer(layer_id)
+                if isinstance(layer, QgsVectorLayer):
+                    layers.append(layer)
+        else:
```

Comment:
do we need this else, since there is one at line 115 and if no layers are passed, theres no layer list passed to identify.

that identify tool code will then use `iface.mapCanvas().layers()` ordering instead, which probably is the "visible" ordering, instead of project order which is unstable (probably added order?), so we dont need to do unnecessary own code here for that?